ccm_web/server/src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,39 @@
+import { Request } from 'express'
+import { Strategy as PassportJwtStrategy } from 'passport-jwt'
+import { Injectable } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { PassportStrategy } from '@nestjs/passport'
+
+import { JwtPayload } from './auth.interfaces'
+import { UserNotFoundError } from '../user/user.errors'
+import { User } from '../user/user.model'
+import { UserService } from '../user/user.service'
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(PassportJwtStrategy) {
+  constructor (
+    private readonly configService: ConfigService,
+    private readonly userService: UserService
+  ) {
+    super({
+      jwtFromRequest: (req: Request): string | null => {
+        if (req.signedCookies.jwt !== undefined) {
+          const token = req.signedCookies.jwt as string
+          return token
+        }
+        return null
```

Comment:
This extracts the JWT token from the signed cookie. I tried to improve the types here (i.e. make TS aware that `request.signedCookies` is an object with certain keys), but I couldn't find a good way to do so.